### ISSUE TITLE
How to run a pretrained model on unlabeled data?

### ISSUE THREAD

**Issue by serenalotreck** [princeton-nlp/PURE#30]
Hi, 

I'm looking to apply your pretrained models to an unlabeled, new dataset. I have my dataset in DyGIE format. Looking at the script, it's unclear to me how to do this, becuase there are only two blocks of code in the script. The first is` if args.do_train:`, where the model is trained, and the second is `if args.do_eval:`, where the model is evaluated. 

I don't want to train, since I'm using a pre-trained model, but I also don't want to evaluate, since my data don't have labels, which makes my use case different than the example of applying the pretrained scibert models to the scierc dataset. 

Wondering if you have pointers on how to do this?

Thanks!

**Comment by a3616001** [comment]
Hi! I guess the easiest way for you to do this is to still create the "ner" and "relations" field in your unlabeled dataset, but set them to be empty for each sentence. For example, if a document contains 4 sentences, you can set the "ner" and "relations" as `{..., "ner": [[], [], [], []], "relations":[[], [], [], []], ...}`. After that, you can use `--do_eval` to generate the prediction file (and ignore the evaluation results in that case).

Thanks for pointing this out! I plan to add a `--do_predict` feature soon. For now, I think this could be an easy way to do only prediction.

**Comment by serenalotreck** [comment]
I just wanted to check in to see if you thought the `--do-predict` feature would be available soon!

**Comment by serenalotreck** [comment]
Just wanted to leave an update for anyone trying this -- your data file should be in a file called `dev.json` -- I originally had mine in `test.json` & couldn't get it to work, but it ran once I changed it to `dev.json`!

Edit: I had typed `test.dev`, but it should be `test.json`

**Comment by Hubotcoder** [comment]
Hi,
allow me to ask a simple question. What is doc_key? According to 'please make sure doc_key can be used to identify a certain document', should I find any document in the sciERC processed data?

**Comment by Shike-Cheng** [comment]
> 你好！我想你做到这一点最简单的方法是仍然在你未标记的数据集中创建“ner”和“relations”字段，但将它们设置为空的每个句子。例如，如果一个文档包含 4 个句子，则可以将“ner”和“relations”设置为`{..., "ner": [[], [], [], []], "relations":[[], [], [], []], ...}`. 之后，您可以使用`--do_eval`生成预测文件（在这种情况下忽略评估结果）。
> 
> 感谢您指出了这一点！我计划很快添加一个`--do_predict`功能。目前，我认为这可能是一种仅进行预测的简单方法。

I would like to know if the prediction function of the model on the unlabeled dataset has been updated, and where I can see the relevant code, thank you very much

**Comment by Shike-Cheng** [comment]
> I just wanted to check in to see if you thought the `--do-predict` feature would be available soon!

I would like to know if the prediction function of the model on the unlabeled dataset has been updated, and where I can see the relevant code, thank you very much

**Comment by Shike-Cheng** [comment]
> Hi! I guess the easiest way for you to do this is to still create the "ner" and "relations" field in your unlabeled dataset, but set them to be empty for each sentence. For example, if a document contains 4 sentences, you can set the "ner" and "relations" as `{..., "ner": [[], [], [], []], "relations":[[], [], [], []], ...}`. After that, you can use `--do_eval` to generate the prediction file (and ignore the evaluation results in that case).
> 
> Thanks for pointing this out! I plan to add a `--do_predict` feature soon. For now, I think this could be an easy way to do only prediction.

I would like to know if the prediction function of the model on the unlabeled dataset has been updated, and where I can see the relevant code, thank you very much
